### PR TITLE
Remove dev-dependencies from badge

### LINKED ISSUE #35: Do not show outdated dev dependencies
Would it make sense to not show outdated `dev-dependencies` on the badge?

It makes the crate seem to be outdated even if it is only for dependencies used to generate tests/benchmarks.

### PR DESCRIPTION
**Pull request by Feliix42**
This PR closes #35 by excluding `dev-dependencies` from the project assessment. An example to test this on can be the `telegram-bot` package:
> http://localhost:8080/crate/telegram-bot/0.7.0

As we have discussed, we may also want to be able to remove build dependencies from the badge, but I figured that this might be something for a different PR.

I will request two reviews, whoever reviews it first may merge afterwards (also feel free to self-request a review!). :)

### CODE REVIEW

**Review by Enet4 — COMMENTED** [review]
Looks OK to me. I would only recommend that `AnalyzedDependencies` is documented somewhere to describe that it currently only considers main and build dependencies.

**Comment by Feliix42** [comment]
You're right. I have added the appropriate docstrings.

File: src/engine/mod.rs [hunk 1]
```diff
@@ -100,12 +100,38 @@ impl AnalyzeDependenciesOutcome {
         self.crates.iter().any(|&(_, ref deps)| deps.any_outdated())
     }
 
+    // TODO(feliix42): Why is this different from the any_outdated() function above?
```
Comment (Enet4) [inline src/engine/mod.rs:103]: ¯\\_(ツ)\_/¯

**Review by Enet4 — APPROVED** [review]
Good docs. :+1:

**Pull request merged**